src/js/relevant.js

```diff
@@ -87,8 +87,15 @@ export default {
              * but currently has 0 repeats, the context will not be available. This same logic is applied in output.js.
              */
             let context = p.path;
-            if ( getChild( node, `.or-repeat-info[data-name="${p.path}"]` ) && !getChild( node,  `.or-repeat[name="${p.path}"]` ) ) {
-                context = null;
+
+            const repeatInfo = getChild( node, `.or-repeat-info[data-name="${p.path}"]` );
+
+            if ( repeatInfo != null && !getChild( node, `.or-repeat[name="${p.path}"]` ) ) {
+                const count = this.form.repeats.updateViewInstancesFromModel( repeatInfo );
```

Comment:
I believe the underlying issue is that the nested repeats are not created when the top-level repeat (no. 2) is created (after the user clicks the plus button). 

The relevant module should not be responsible for fixing that, as it's the responsibility of the repeat module. It's odd that something so elementary is not done correctly. I wonder if this bug was introduced when we started allowing 0 repeat instances. There seems to be some (still mysterious) special condition in the test form, because in e.g. http://localhost:8005/?xform=nested_repeats.xml the bug doesn't surface.

Reply:
I was hesitant to apply this fix here too, I'm glad you agree. To be honest I chose this path because I struggled to understand the repeat initialization flow and wasn't sure how to address it there. But I'll take another look because I agree it would be better solved there.